test/parallel/test-http-outgoing-buffer.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const http = require('http');
+const OutgoingMessage = http.OutgoingMessage;
+
+const msg = new OutgoingMessage();
+
+// writes should be buffered until highwatermark
+// even when no socket is assigned.
+
+assert.strictEqual(msg.write('asd'), true);
+while (msg.write('asd'));
+assert.strictEqual(msg.write('asd'), false);
```

Comment:
What do think about replacing this with something like `assert(msg.outputSize >= HIGH_WATER_MARK)`. Return value is already tested with above.